tests/sagemaker-tests/test_sagemaker_utils.py

```diff
@@ -221,3 +233,156 @@ def test_wrap_model_artifacts():
+    error_response = {'Error': {'Code': '404'}}
+    mock_s3.head_object.side_effect = ClientError(error_response, 'HeadObject')
+    with pytest.raises(AssertionError, match='not a compressed tar file'):
+        check_tarfile_s3_object('s3://graphstorm-demos/test/GSF-demo.pptx')
+
+@patch('launch_utils.S3Uploader.upload')
+def test_upload_data_to_s3(mock_s3uploader):
+    """ Test the upload data to S3 function.
+    """
+    mock_s3uploader.return_value = 's3://graphstorm-demos/test/model.tar.gz'
+    
+    # Test case 1: mock successful upload
+    ret = upload_data_to_s3('s3://graphstorm-demos/test/', './model.tar.gz', 'session')
+    mock_s3uploader.assert_called_once_with('./model.tar.gz', 's3://graphstorm-demos/test/',
+                                            sagemaker_session='session')
+
+    assert ret == 's3://graphstorm-demos/test/model.tar.gz'
+
+    # Test case 2: mock unsucessful upload
+    ret = upload_data_to_s3('s3://graphstorm-demos/test/', './model.tar.gz', 'session')
+    with pytest.raises(AssertionError):
+        mock_s3uploader.assert_called_once_with('s3://graphstorm-demos/test/', './model.tar.gz',
+                                                sagemaker_session='session')
+    
+
+def local_test_check_tarfile_s3_object(input_value, expected_result):
+    """ Test the check if a model tarfile exist in the given S3 url location.
+
+    ## This test should NOT be ran in CI pipeline, but in local environment only! ##
+    
+    Testers need to provide an S3 url as the input value, and True or False as the
+    expected result in the python command line.
+
+    """
+    s3_url = input_value
+
+    if expected_result.lower() == 'true':
+        assert check_tarfile_s3_object(s3_url)
+    else:
+        assert not check_tarfile_s3_object(s3_url)
+
+
+def local_test_upload_data_to_s3(input_value, expected_result=None):
+    """ Test the check if can upload a file to the given S3 url location.
+
+    ## This test should NOT be ran in CI pipeline, but in local environment only! ##
+    
+    Testers need to provide an S3 url as the input value. This test function will create
+    a temporary emtpy file as the data to be uploaded.
+    """
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        data_path = os.path.join(tmpdirname, 'data.txt')
+        create_dummy_file(data_path)
+
+        # prepare sessions
+        b3_session = boto3.Session(region_name='us-west-2')
+        sm_session = sm.Session(boto_session=b3_session)
+
+        s3_path = input_value
+        ret = upload_data_to_s3(s3_path, data_path, sm_session)
+
+    expected_result = input_value + '/data.txt'
+    assert ret == expected_result
+
+    # delete the uploaded object
+    parsed_url = urlparse(ret)
+    bucket_name = parsed_url.netloc
+    object_key = parsed_url.path.lstrip('/')
+    s3 = boto3.client('s3')
+    s3.delete_object(Bucket=bucket_name, Key=object_key)
+
+
+# if __name__ == '__main__':
```

Comment:
please clean up.

Reply:
This is purposely left commented for local tests.